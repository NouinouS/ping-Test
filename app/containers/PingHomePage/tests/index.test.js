/**
 *
 * Tests for PingHomePage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';
import { PingHomePage, mapDispatchToProps } from '../index';


describe('<PingHomePage />', () => {

  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <PingHomePage loadAddrs={()=>{}} recentAddrs={['192.168.1.1','1.1.1.1']} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the localStorage on mount', () => {
    const loadSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <PingHomePage
            loadAddrs={loadSpy}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(loadSpy).toHaveBeenCalled();
  });

});
