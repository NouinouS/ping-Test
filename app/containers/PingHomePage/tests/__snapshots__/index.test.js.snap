// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PingHomePage /> should render and match the snapshot 1`] = `
<div>
  <span>
    This is the PingHomePage container!
  </span>
  <div
    class="Wrapper-sc-1umgotm-0 laCwDT"
  >
    <ul
      class="Ul-mwnq6h-0 jcxXFy"
    >
      <div>
        <button>
          Add
        </button>
        192.168.1.1
      </div>
      <div>
        <button>
          Add
        </button>
        1.1.1.1
      </div>
    </ul>
  </div>
</div>
`;
